near-contract-standards/src/event.rs

```diff
@@ -0,0 +1,299 @@
+    #[serde(borrow)]
+    pub owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub token_ids: Vec<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub authorized_id: Option<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub memo: Option<Cow<'a, str>>,
+}
+
+impl<'a> NftBurnData<'a> {
+    pub fn new<S>(
+        owner_id: S,
+        token_ids: Vec<S>,
+        authorized_id: Option<S>,
+        memo: Option<S>,
+    ) -> NftBurnData<'a>
+    where
+        S: Into<Cow<'a, str>>,
+    {
+        Self {
+            owner_id: owner_id.into(),
+            token_ids: token_ids.into_iter().map(|s| s.into()).collect(),
+            authorized_id: authorized_id.map(|t| t.into()),
+            memo: memo.map(|t| t.into()),
+        }
+    }
+}
+
+impl<'a> NearEvent<'a> {
+    pub fn new_171(version: &'static str, event_kind: Nep171EventKind<'a>) -> Self {
+        NearEvent::Nep171(Nep171Event { version, event_kind })
+    }
+
+    pub fn new_171_v1(event_kind: Nep171EventKind<'a>) -> Self {
+        NearEvent::new_171("1.0.0", event_kind)
+    }
+
+    pub fn nft_burn(data: Vec<NftBurnData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftBurn(data))
+    }
+    pub fn nft_transfer(data: Vec<NftTransferData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftTransfer(data))
+    }
+
+    pub fn nft_mint(data: Vec<NftMintData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftMint(data))
+    }
+
+    pub(crate) fn to_json_string(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+
+    pub fn to_json_event_string(&self) -> String {
+        format!("EVENT_JSON:{}", self.to_json_string())
+    }
+
+    pub fn emit(&self) {
+        near_sdk::env::log_str(&self.to_json_event_string());
+    }
+
+    pub fn emit_nft_mint(owner_id: &str, token_ids: Vec<&str>, memo: Option<&str>) {
+        NearEvent::emit_nft_mints(vec![NftMintData::new(owner_id, token_ids, memo)]);
+    }
+
+    pub fn emit_nft_mints(data: Vec<NftMintData<'a>>) {
+        NearEvent::nft_mint(data).emit();
+    }
+
+    pub fn emit_nft_transfer(
+        old_owner_id: &str,
+        new_owner_id: &str,
+        token_ids: Vec<&str>,
+        authorized_id: Option<&str>,
+        memo: Option<&str>,
+    ) {
+        NearEvent::emit_nft_transfers(vec![NftTransferData::new(
+            old_owner_id,
+            new_owner_id,
+            token_ids,
+            authorized_id,
+            memo,
+        )]);
+    }
+
+    pub fn emit_nft_transfers(data: Vec<NftTransferData<'a>>) {
+        NearEvent::nft_transfer(data).emit()
+    }
+
+    pub fn emit_nft_burn(
+        owner_id: &str,
+        token_ids: Vec<&str>,
+        authorized_id: Option<&str>,
+        memo: Option<&str>,
+    ) {
+        NearEvent::emit_nft_burns(vec![NftBurnData::new(owner_id, token_ids, authorized_id, memo)]);
+    }
+
+    pub fn emit_nft_burns(data: Vec<NftBurnData<'a>>) {
+        NearEvent::nft_burn(data).emit()
+    }
```

Comment:
I feel these `emit_` wrappers will only confuse and unnecessarily introduce complexity and the number of methods a user should learn about.

Let's remove these.